cli/src/main/java/hudson/cli/FullDuplexHttpStream.java

```diff
@@ -48,16 +48,31 @@ private static String basicAuth(String userInfo) {
         return null;
     }
 
+    @Deprecated
+    public FullDuplexHttpStream(URL target, String authorization) throws IOException {
+        this(new URL(target.toString().replaceFirst("/cli.*$", "")), target.toString().replaceFirst("(.+)/cli.*$", "$1"), authorization);
```

Comment:
Yeah the original code in `createCrumbUrlBase` was also too clever for its own good. I can add an explanatory comment.